peripherals/rcc/rcc_u5.yaml

```diff
@@ -0,0 +1,721 @@
+      Enabled: [1, "Interrupt enabled"]
+  CIFR:
+    "*RDYF":
+      _read:
+        NotInterrupted: [0, "No clock ready interrupt"]
+        Interrupted: [1, "Clock ready interrupt"]
+  CICR:
+    "*RDYC":
+      _write:
+        Clear: [1, "Clear flag"]
+  A[HP]B?RSTR*:
+    "*RST":
+      NoEffect: [0, "No effect"]
+      Reset: [1, "Reset peripheral"]
+  A[HP]B?ENR*:
+    "*EN":
+      Disabled: [0, "Peripheral clock disabled"]
+      Enabled: [1, "Peripheral clock enabled"]
+  A[HP]B?SMENR*:
+    "*SMEN":
+      Disabled: [0, "Peripheral clocks disabled by the clock gating during Sleep and Stop modes"]
+      Enabled: [1, "Peripheral clocks enabled by the clock gating during Sleep and Stop modes"]
+  SRDAMR:
+    "*AMEN":
+      Disabled: [0, "Peripheral autonomous mode disabled during Stop 0/1/2 mode"]
+      Enabled: [1, "Peripheral autonomous mode enabled during Stop 0/1/2 mode"]
+  CCIPR1:
+    TIMICSEL:
+      Disabled: [0, "HSI, MSIK and MSIS dividers disabled"] # same for [0, 3]
+      HsiMsisMsis: [4, "HSI/256, MSIS/1024 and MSIS/4 generated and can be selected by TIM16, TIM17, and LPTIM2 as internal input capture"]
+      HsiMsisMsik: [5, "HSI/256, MSIS/1024 and MSIK/4 generated and can be selected by TIM16, TIM17, and LPTIM2 as internal input capture"]
+      HsiMsikMsis: [6, "HSI/256, MSIK/1024 and MSIS/4 generated and can be selected by TIM16, TIM17, and LPTIM2 as internal input capture"]
+      HsiMsikMsik: [7, "HSI/256, MSIK/1024 and MSIK/4 generated and can be selected by TIM16, TIM17, and LPTIM2 as internal input capture"]
+    ICLKSEL:
+      HSI: [0, "HSI48 clock selected"]
+      PLL2Q: [1, "PLL2 \"Q\" (pll2_q_ck) selected"]
+      PLL1Q: [2, "PLL1 \"Q\" (pll1_q_ck) selected"]
+      MSIK: [3, "MSIK clock selected"]
+    FDCAN1SEL:
+      HSE: [0, "HSE clock selected"]
+      PLL1Q: [1, "PLL1 \"Q\" (pll2_q_ck) selected"]
+      PLL2P: [2, "PLL2 \"P\" (pll1_p_ck) selected"]
+    SYSTICKSEL:
+      HCLK_Div8: [0, "HCLK/8 selected"]
+      LSI: [1, "LSI selected"]
+      LSE: [2, "LSE selected"]
+    LPTIM2SEL:
+      PCLK1: [0, "PCLK1 selected"]
+      LSI: [1, "LSI selected"]
+      HSI16: [2, "HSI16 selected"]
+      MSIK: [3, "MSIK selected"]
+    SPI2SEL,I2C[124]SEL,UART[45]SEL,USART[23]SEL:
+      PCLK1: [0, "PCLK1 selected"]
+      SYSCLK: [1, "SYSCLK selected"]
+      HSI16: [2, "HSI16 selected"]
+      MSIK: [3, "MSIK selected"]
+    SPI1SEL,USART1SEL:
+      PCLK2: [0, "PCLK2 selected"]
+      SYSCLK: [1, "SYSCLK selected"]
+      HSI16: [2, "HSI16 selected"]
+      MSIK: [3, "MSIK selected"]
+  CCIPR2:
+    OTGHSSEL:
+        HSE: [0, "HSE selected"]
+        PLL1P: [1, "PLL1 \"Q\" (pll1_q_ck) selected"]
+        HSE2: [2, "HSE/2 selected"]
+        PLL1P_Div2: [3, "PLL1 \"P\" divided by 2 (pll1_p_ck/2) selected"]
+    I2C[56]SEL,USART6SEL:
+      PCLK1: [0, "PCLK1 selected"]
+      SYSCLK: [1, "SYSCLK selected"]
+      HSI16: [2, "HSI16 selected"]
+      MSIK: [3, "MSIK selected"]
```

Comment:
```suggestion
    I2C[56]SEL,USART6SEL:
      _derivedFrom: RCC.CCIPR1.SPI1SEL.SPI1SEL
```

Reply:
I presume you meant `derivedFrom` without the underscore? But this syntax doesn't seem to work for me. I added `_name: ClkSelSpiUart` to the other one, but I am getting an error
```
    Caused by:
        0: Processing device `STM32U5A9`
        1: According to `RCC`
        2: Processing peripheral `RCC`
        3: According to `CCIPR2`
        4: Processing register `CCIPR2`
        5: Processing field matched to `I2C[56]SEL,USART6SEL`
        6: Adding read-write enumeratedValues
        7: Value is not a vector (array): String("RCC.CCIPR1.USART1SEL.ClkSelSpiUart")
```

Reply:
No I meant with underscore: https://github.com/rust-embedded/svdtools/blob/1426fdecb1af98020c62a3e58ec0964f59bd3c6e/src/patch/register.rs#L735 . But there no implementation for crossregister enum derive.
But you could derive whole field. It is even better:
```yaml
  CCIPR2:
    _modify:
      I2C?SEL,USART?SEL:
        derivedFrom: RCC.CCIPR1.USART1SEL
```

Reply:
This does not throw any errors, but the patched .svd is missing the enumeratedValues for `I2C5SEL`, `I2C6SEL` and `USART6SEL`. I will push the other suggestions for now, let me know if you can get it working.

Reply:
Works as expected for me:
```xml
            <field derivedFrom="RCC.CCIPR1.USART1SEL">
              <name>USART6SEL</name>
              <description>USART6 kernel clock source selection
These bits are used to select the USART6 kernel clock source.
The USART6 is functional in Stop 0 and Stop 1 modes only when the kernel clock is HSI16 or LSE.
Note: This bitfield is only available on some devices in the STM32U5 Series. Refer to the device datasheet for availability of its associated peripheral. If not present, consider this bitfield as reserved and keep it at reset value.</description>
              <bitOffset>16</bitOffset>
              <bitWidth>2</bitWidth>
              <access>read-write</access>
            </field>
```
It should not content new `enumeratedValues` section as it derives existent one.

Reply:
Ah, I see now. I have added that change 👍